### PR TITLE
[feature](Nereids) make count() avaliable for all data type

### PR DESCRIPTION
**Pull request by keanji-x**
## Proposed changes

make count() avaliable for all data type

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/Count.java [hunk 1]
```diff
@@ -82,11 +81,8 @@ public void checkLegalityBeforeTypeCoercion() {
     public void checkLegalityAfterRewrite() {
         // after rewrite, count(distinct bitmap_column) should be rewritten to bitmap_union_count(bitmap_column)
         for (Expression argument : getArguments()) {
-            if (argument.getDataType().isObjectType()) {
-                throw new AnalysisException(Type.OnlyObjectTypeErrorMsg);
-            }
-            if (distinct && argument.getDataType().isComplexType()) {
-                throw new AnalysisException("COUNT DISTINCT could not process complex type " + this.toSql());
+            if (distinct && (argument.getDataType().isComplexType() || argument.getDataType().isObjectType())) {
```
[the remark this answers is not in the document]
Comment (XieJiann) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/Count.java]: not support.  has added.

**Comment by XieJiann** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by jackwener — APPROVED** [review]

**Comment by xiaokang** [comment]
can not be merged to 2.0 since complex datatypes is not supported

**Pull request merged**